packages/web/src/js/Tooltip.ts

```diff
@@ -109,6 +166,132 @@ class Tooltip extends BaseComponent {
 
     return this.tip;
   }
+
+  getTooltipFloatingProps() {
+    const { flip, shift, size, flipCrossAxis, flipFallbackAxisSideDirection, flipFallbackPlacements, placement } = this
+      .config as Config;
+
+    return {
+      placement,
+      flip: flip ?? false,
+      shift: shift ?? false,
+      size: size ?? false,
+      flipCrossAxis: flipCrossAxis ?? false,
+      flipFallbackPlacements: flipFallbackPlacements && transformStringToArray(flipFallbackPlacements),
+      flipFallbackAxisSideDirection: flipFallbackAxisSideDirection || 'none',
+    };
+  }
+
+  getFlipConfig() {
+    const { flip } = FloatingUI;
+    const tooltipFloatingProps = this.getTooltipFloatingProps();
+    const {
+      flip: flipProp,
+      flipCrossAxis,
+      flipFallbackAxisSideDirection,
+      flipFallbackPlacements,
+    } = tooltipFloatingProps;
+
+    return (
+      flipProp &&
+      flip({
+        mainAxis: flipProp,
+        crossAxis: flipCrossAxis,
+        fallbackAxisSideDirection: flipFallbackAxisSideDirection,
+        /* eslint-disable no-undefined */
+        fallbackPlacements: flipFallbackPlacements || undefined,
+      })
+    );
+  }
+
+  getShiftConfig() {
+    const { shift, limitShift } = FloatingUI;
+    const tooltipFloatingProps = this.getTooltipFloatingProps();
+
+    return (
+      tooltipFloatingProps.shift &&
+      shift({
+        limiter: limitShift({
+          offset: ({ rects }) => ({
+            mainAxis: rects.reference.height,
+          }),
+        }),
+      })
+    );
+  }
+
+  getSizeConfig(tooltip: HTMLElement) {
+    const { size } = FloatingUI;
+    const { tooltipMaxWidth } = this;
+    const tooltipFloatingProps = this.getTooltipFloatingProps();
+
+    return (
+      tooltipFloatingProps.size &&
+      size({
+        apply({ availableWidth }: { availableWidth: number }) {
+          Object.assign(tooltip.style, {
+            maxWidth: `${tooltipMaxWidth < availableWidth ? tooltipMaxWidth : availableWidth}px`,
+          });
+        },
+      })
+    );
+  }
+
+  updateTooltipPosition(button: HTMLElement, tooltip: HTMLElement) {
+    const { computePosition, offset: floatingOffset, arrow: floatingArrow } = FloatingUI;
+    const { tooltipOffset, arrowCornerOffset, arrow } = this;
+    const { placement: userPlacement } = this.getTooltipFloatingProps();
+
+    const hasFlip = this.getFlipConfig();
+    const hasShift = this.getShiftConfig();
+    const hasSize = this.getSizeConfig(tooltip);
+
+    if (!button || !tooltip) return;
```

Comment:
Can we reverse the condition, please? After the modification, the `return` will not be needed.

Reply:
condition reversed